sw/device/silicon_creator/lib/drivers/spi_device.c

```diff
@@ -622,8 +622,9 @@ void spi_device_cmd_get(spi_device_cmd_t *cmd) {
         abs_mmio_read32(kBase + SPI_DEVICE_UPLOAD_ADDRFIFO_REG_OFFSET);
   }
 
+  // TODO: Handle the case of non-zero PAYLOAD_START_IDX.
   cmd->payload_byte_count =
-      bitfield_field32_read(reg, SPI_DEVICE_UPLOAD_STATUS_PAYLOAD_DEPTH_FIELD);
+      bitfield_field32_read(reg, SPI_DEVICE_UPLOAD_STATUS2_PAYLOAD_DEPTH_FIELD);
```

Comment:
@alphan Just revised the reference of PAYLOAD_DEPTH field.
Still it needs to process the non-zero PAYLOAD_START_IDX (in case of payload is more than 256B)

Reply:
CC: @arunthomas